ece595_project_2/source/ultrasonic.c

```diff
@@ -0,0 +1,89 @@
+/*
+ * ultrasonic.c
+ *
+ *  Created on: Apr 6, 2019
+ *      Author: Lacy
+ */
+#include "fsl_ftm.h"
+#include "clock_config.h"
+#include "fsl_port.h"
+#include "pin_mux.h"
+#include "io_abstraction.h"
+#include "ultrasonic.h"
+#include "time_schedule.h"
+#include "http_client_app.h"
+#include "proximity_estimation.h"
+#include "ultrasonic.h"
+
+/* Get source clock for FTM driver */
+#define FTM_SOURCE_CLOCK (CLOCK_GetFreq(kCLOCK_BusClk)/1)
+#define DOGCOUNT 4
+
+static Proximity_Status_T Prox_Status;
+static Time_Schedule_Status_T Time_Status;
+static Weather_Status_T Weather_Status;
+static volatile Dog_Status_T dog_status;
+static uint8_t dog_counter;
+
+void Init_USS_Timer(void)
+{
+    ftm_config_t ftm_info;
+    FTM_GetDefaultConfig(&ftm_info);
+    /* Divide FTM clock by 4 */
+    ftm_info.prescale = kFTM_Prescale_Divide_4;
+    FTM_Init(FTM1, &ftm_info);
+    FTM_SetTimerPeriod(FTM1, USEC_TO_COUNT(1U, FTM_SOURCE_CLOCK));
+}
+
+void Detect_Pet(void)
+{
+	Time_Status = Get_Time_Schedule_Status();
+	Weather_Status = Get_Weather_Status();
+	Prox_Status = Get_Proximity_Status();
+	if(Time_Status == NOT_RESTRICTED && Weather_Status == GOOD_WEATHER && Prox_Status == CLOSE)
```

Comment:
It doesn't need to be fixed, but it's typically good practice to put the constant on the left. This will ensure that single "=" is flagged by the compiler.